packages/build/tests/pre_dev_cleanup/tests.js

```diff
@@ -0,0 +1,39 @@
+import { access } from 'node:fs/promises'
+import { join } from 'path'
+
+import { Fixture } from '@netlify/testing'
+import test from 'ava'
+
+test('Build removes blobs directory before starting', async (t) => {
+  const fixture = await new Fixture('./fixtures/with_preexisting_blobs')
+    .withFlags({ debug: false, timeline: 'dev' })
+    .withCopyRoot()
+
+  const blobsDir = join(fixture.repositoryRoot, '.netlify', 'deploy', 'v1', 'blobs', 'deploy')
+
+  await t.notThrowsAsync(access(blobsDir))
+
+  const output = await fixture.runDev(() => {})
+
+  t.true(output.includes('Cleaning up leftover files from previous builds'))
+  t.true(output.includes(`Cleaned up .netlify/deploy/v1/blobs/deploy`))
```

Comment:
Looking for this string in the output feels a bit brittle. Not only you're not actually testing if the files were deleted, but the test will start failing if we change the output.

Have you considered looking for the actual files on disk? We do that in some tests, like: https://github.com/netlify/build/blob/ea09991f2b093cf0adb6441a77d6972bd8360736/packages/build/tests/edge_functions/tests.js#L195

Reply:
I got this pattern in other tests, should I clear them up from those files as well?

Right below this line the test checks that the directory is gone `await t.throwsAsync(access(blobsDir))`, is that sufficient, or would using `pathExists` be preferred

Reply:
Ah, I hadn't noticed the `access()` call. I think that would be sufficient, and I think we can just remove the assertions on `output` (but if you want to keep them there, no strong objections from me).